graylog2-server/src/main/java/org/graylog2/rest/resources/system/logs/LoggersResource.java

```diff
@@ -151,6 +146,43 @@ public void setSingleLoggerLevel(
         logger.setLevel(newLevel);
     }
 
+    @GET
+    @Timed
+    @ApiOperation(value = "Get recent log messages")
+    @ApiResponses(value = {
+            @ApiResponse(code = 404, message = "Memory appender not configured.")
+    })
+    @Path("/messages/recent")
+    @Produces(MediaType.APPLICATION_JSON)
+    public LogMessagesSummary messages(@ApiParam(name = "max", required = true) @QueryParam("max") int max) {
+        List<LogMessage> messages = new ArrayList<>();
+
+        Logger logger = Logger.getRootLogger();
+        if (logger.getAppender("memory") == null) {
+            throw new WebApplicationException(404);
+        }
+
+        MemoryAppender appender = (MemoryAppender) logger.getAppender("memory");
+        for (LoggingEvent loggingEvent : appender.getLogMessages(max)) {
+            List<String> throwable = new ArrayList<>();
```

Comment:
Why are you initializing a new ArrayList here just to overwrite it soon after? If it's for the purpose of using an empty list in case of ``getThrowableStrRep()`` returning null, then doing that in an else-clause and using ``Collections.emptyList()`` would be better.